ui/job-view/context/PinnedJobs.jsx

```diff
@@ -0,0 +1,177 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { thPinboardCountError, thPinboardMaxSize } from '../../js/constants';
+
+const PinnedJobsContext = React.createContext({});
+
+export class PinnedJobs extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      pinnedJobs: {},
+      pinnedJobBugs: {},
+      isPinBoardVisible: false,
+    };
+  }
+
+  componentDidMount() {
+    this.pinJob = this.pinJob.bind(this);
```

Comment:
The React docs only ever seem to talk about the constructor when discussing binding the event handlers:
https://reactjs.org/docs/handling-events.html
https://reactjs.org/docs/react-component.html#constructor

Should these be in the constructor instead of `componentDidMount`?

Reply:
I believe this is something that's odd because we're partially Angular.  Yeah, it always says to put these ``bind``s in the constructor.  And sometimes they DO work there, but sometimes not.  I'm hopeful I can move them all into the constructors once we don't have to use ``react2angular`` anymore.